src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -381,6 +385,16 @@ protected CompletableFuture<FetchedValue> fetchField(ExecutionContext executionC
     }
 
     private CompletableFuture<FetchedValue> fetchField(GraphQLFieldDefinition fieldDef, ExecutionContext executionContext, ExecutionStrategyParameters parameters) {
+
+        int resultNodesCount = executionContext.getResultNodesCount().incrementAndGet();
+        Integer maxNodes;
+        if ((maxNodes = executionContext.getGraphQLContext().get(MAX_RESULT_NODES)) != null) {
+            if (resultNodesCount > maxNodes) {
+                executionContext.getGraphQLContext().put(MAX_RESULT_NODES_BREACHED, true);
+                return CompletableFuture.completedFuture(new FetchedValue(null, Collections.emptyList(), null));
```

Comment:
whats the point  of the new MAX_RESULT_NODES_BREACHED value?

Reply:
Should we start returning null and hence risk all the non null  validation or JUST throw AbortExectionException ?  Stop harder!!!

Reply:
My suggestion

* Create a `ExecutedNodes` class
* Put the count in there regardless if its breached
* Put this in at the end of the request
* Have a "maxNodesExceeded" boolean say and set that here
*

Reply:
done